### PR TITLE
Add "Content-Security-Policy" to speculation rules explainer

### PR DESCRIPTION
**Pull request by toyoshim**
The new section explains how the prerender and prefetch follow the Content-Security-Policy. They respect the existing `script-src`, but `inline-speculation-rules` is also added as a new source to permit inline speculationrules.

### CODE REVIEW

**Comment by toyoshim** [comment]
This is a revised proposal that is based on discussion at #209

File: triggers.md [hunk 1]
```diff
@@ -283,6 +284,13 @@ Adding `"dns-prefetch"` and `"preconnect"`, to mirror [Resource Hints](https://w
 
 Another envisioned speculative action is `"prefetch_with_subresources"`, which prefetches a document and then uses the HTML preload scanner to find other subresources that are worth preloading. Chromium currently does something similar (known as "[NoState Prefetch](https://developer.chrome.com/blog/nostate-prefetch/)") for `<link rel="prerender">`. But, we're not yet sure this feature is pulling its weight, in between the lightweight prefetch and the fully-instant prerender features, so it's not yet clear whether this will be worth integrating.
 
+### Content-Security-Policy
+
+`Speculation-Rules` is embedded as an inline declarationt within a script tag with type=”speculationrules”, and restricted by the `script-src` restriction of the `Content-Security-Policy`.
```
Comment (jeremyroman) [inline triggers.md:289]:
```suggestion
Speculation rules can be embedded inline within a script tag with type=”speculationrules”, and restricted by the `script-src` CSP directive.
```

The HTTP header name isn't really applicable here, so we can just describe it as "speculation rules".

File: triggers.md [hunk 2]
```diff
@@ -283,6 +284,13 @@ Adding `"dns-prefetch"` and `"preconnect"`, to mirror [Resource Hints](https://w
...
+To control the availability to inject an inline `Speculation-Rules`, `'inline-speculation-rules'` is available in addition to `‘unsafe-inline’`.
```
Comment (jeremyroman) [inline triggers.md:290]:
```suggestion
To allow inline speculation rules, use either the `'inline-speculation-rules'` or `‘unsafe-inline’` keyword.
```

File: triggers.md [hunk 3]
```diff
@@ -283,6 +284,13 @@ Adding `"dns-prefetch"` and `"preconnect"`, to mirror [Resource Hints](https://w
...
+This helps developers to permit only inline `Speculation-Rules` but still to disallow unsafe inline JavaScript.
```
Comment (jeremyroman) [inline triggers.md:291]:
```suggestion
Using `script-src 'inline-speculation-rules'` helps developers to permit inline speculation rules but still disallow unsafe inline JavaScript.
```

File: triggers.md [hunk 4]
```diff
@@ -283,6 +284,13 @@ Adding `"dns-prefetch"` and `"preconnect"`, to mirror [Resource Hints](https://w
...
+`prefetch-src` is also used to restrict target URLs.
```
Comment (jeremyroman) [inline triggers.md:292]:
```suggestion
The `prefetch-src` directive can be used to restrict which URLs can be prefetched or prerendered.
```

**Review by jeremyroman — APPROVED** [review]
A few suggested tweaks to working, but lgtm otherwise.

**Review by domenic — APPROVED** [review]
LGTM with Jeremy's suggested fixes. (I also edited Jeremy's suggested fixes to change some curly quotes to straight quotes.)

**Comment by toyoshim** [comment]
Thanks!
This is my first code review at the GitHub system and I might make a mistake. But I think I merged @jeremyroman 's suggestions into the pull request, and this PR is now ready for the 2nd review or merging.
Please let me know if I still need some actions.

**Pull request merged**